cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -132,3 +133,228 @@ def test_tags(provider):
     assert len(deps_db) > 0, "There is no deployment(s) available in UI"
     deployment = get_random_list(deps_db, 1)[0]
     deployment.validate_tags(tags=tags)
+
+
+@pytest.mark.parametrize("archive", [RESOURCE_WAR_PATH, RESOURCE_JAR_PATH, RESOURCE_EAR_PATH])
+def test_deploy(provider, archive):
+    """Tests Deployment of provided archive into EAP7 server
+
+    Steps:
+        * Get servers list from UI
+        * Chooses JBoss EAP server from list
+        * Invokes 'Add Deployment' toolbar operation
+        * Selects "war" file to upload.
+        * Chose random Runtime Name.
+        * Checks that notification message is shown.
+        * Refreshes the provider.
+        * Verifies that deployment is shown in list and is Enabled.
+        * Selects deployment to show the details.
+        * Verified details properties.
+    """
+    server = _get_eap_server(provider)
+    _deploy(provider, server, archive)
+    # disable for now as not stable test because of performance issue #10138
+    # _check_deployment_enabled(provider, server, runtime_name)
+    # deployment = _get_deployment_from_list(provider, server, runtime_name)
+    # deployment.validate_properties()
+
+
+# disable for now as not stable test because of performance issue #10138
+@pytest.mark.uncollect
+@pytest.mark.parametrize("archive", [RESOURCE_WAR_PATH, RESOURCE_JAR_PATH, RESOURCE_EAR_PATH])
+def test_undeploy(provider, archive):
+    """Tests Undeployment of archive from EAP7 server
+
+    Steps:
+        * Get servers list from UI
+        * Chooses JBoss EAP server from list
+        * Deplays some deployment archive into server.
+        * Select that deployment from deployments list.
+        * Performs "Undeploy" toolbar operation on it.
+        * Refreshes the provider.
+        * Lists all deployments on EAP server.
+        * Verified the recently undeployed archive's status is Disabled.
+        * Selects that archive from the list to load details.
+        * Verifies that properties of deplyment's summary page and the status is Disabled.
+    """
+    server = _get_eap_server(provider)
```

Comment:
Tests should be designed so they use the "public" interfaces of objects.

*I say "public" in quotes because everything is public in python :smile: *